subprojects/docs/src/snippets/dependencyManagement/catalogs-asprovider/groovy/build.gradle

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id 'scala'
+}
+
+// tag::use_version_asprovider[]
+// getting the version of 'zinc' requires `asProvider()` when there are other 'zinc-...' versions:
```

Comment:
The comment feels redundant. If you want to keep it, move it to being a callout in the snippet, with a reference under its inclusion.
And of course, while searching an example to reference, I realize this whole doc section has been using comments ...
Then I would move it right above the line that uses the construct instead of above the block